translate: basic/*-sorts (#124)
OI-wiki/i18n: docs/basic/tournament-sort.md

Last translate with upstream: [08eb0c5](https://github.com/OI-wiki/OI-wiki/commit/08eb0c53b1ad55b518b0f5f7a569c25b8c1215b2#diff-ba0721b5c036c9f0301e88f3247a0b128f22996796cd5526d5d6220ef53d01b3) [#3333](https://github.com/OI-wiki/OI-wiki/pull/3333) on Jul 17, 2021.

This article will briefly introduce the tournament sort.

## Introduction

> Tournament sort<!---, or tree selection sort,--> is a optimization for naive [selection sort](./selection-sort.md), and a variation of [heapsort](./heap-sort.md) (with complete ). It performs based by selection sort by using a priority queue to find next element to select.

Reviewer: aka?
为啥注释掉了（

Author: 同样是因为翻不到这么表述的原因
will fix to aka

> The algorithm is named after a single-elimination tournament, where there are many players (or teams) that play in two-sided matches. Each much compares the players and the winning one is promoted to next round. The tournament determines the best player, but the player who was beaten in the final match may not be the second best – he may be inferior to other players the winner bested.[^ref1] <!---原文于此处部分引用了Wikipedia对其构词学的描述, 故加上ref --> 

Reviewer: ```suggestion
The algorithm is named after a single-elimination tournament, where there are many players (or teams) that play in two-sided matches. Each match compares the players and the winner is promoted to next round. The tournament determines the best player, but the player who was beaten in the final match may not be the second best – he may be inferior to other players the winner beated.[^ref1]
```

> ## Working Theory

Reviewer: 统一用词

Author: Fixed in 657c496b

> Here is an illustration of **minimum tournament tree** for example:

Reviewer: ```suggestion
Here is an example of **minimum tournament tree**:
```

![tournament-sort1](./images/tournament-sort1.png)

The leaf nodes represent unordered elements. The red edges represent winner path of each round. Obviously, finishing a tournament can find out the minimum element from a group of elements.

After each round of comparison among $n$ elements, we will have $\frac{n}{2}$ "winners". The smaller one from each pair of elements will enter next round of comparison. If no pair could be made, this element will directly enter next round.

![tournament-sort2](./images/tournament-sort2.png)

> After one "tournament" we need to eliminate the winner element by setting it to $\infty$ (which is similar to [heapsort](./heap-sort.md))and hold the "tournament" again to select secondly small element.

Reviewer: ```suggestion
After one "tournament" we need to eliminate the winner element by setting it to $\infty$ (which is similar to [heapsort](./heap-sort.md) and hold the "tournament" again to select secondly small element.
```
